osu.Game.Rulesets.Catch/Mods/CatchModRelax.cs

```diff
@@ -20,7 +18,7 @@ public class CatchModRelax : ModRelax, IApplicableToDrawableRuleset<CatchHitObje
     {
         public override string Description => @"Use the mouse to control the catcher.";
 
-        private DrawableRuleset<CatchHitObject> drawableRuleset;
+        private DrawableRuleset<CatchHitObject> drawableRuleset = null!;
```

Comment:
Probably want to mark this as nullable and assert in places where it can't be null. 

`null!` should never be used unless there's no way for the field to be accessed while it has a null value.

Reply:
Not sure. We generally use `null!` for cases like this where it's "guaranteed after a known point in time", ie. `LoadComplete` or `load()` or in this case `ApplyToDrawableRuleset`. I think it might be okay to leave it?

Reply:
I was about to post the same comment as @peppy. It's somewhat like dependency injection, where you can reasonably assume it's non-null based on the order in which methods run, and `Playfield`s are not meant to be (or maybe even _cannot_) instantiated without a `DrawableRuleset`.

Reply:
Would be fine with using `AsNonNull()` in this case.

Reply:
I'd prefer to mark the `drawableRuleset` as nullable and add the null check in the `ApplyToPlayer` (follow @frenzibyte's suggestion).
As @smoogipoo mention, we can not make sure that `ApplyToPlayer` is always after `ApplyToDrawableRuleset` in in non-gameplay contexts.

Also, `ApplyToPlayer()` seems required `drawableRuleset` should not be null, so maybe `Debug.Assert()` might be more suitable?

Reply:
My suggestion was to assert, not null check, and at this point there's no difference between `Debug.Assert` and `AsNonNull` (one will throw assertion exception, the other will just nullref), so it's shorter to use the latter.